spec/lib/repo/git_revision_spec.rb

```diff
@@ -3,6 +3,52 @@
 describe Repository::GitRevision do
   context 'with a git repo' do
 
+    before(:context) do
+
+      # Make sure gitolite-admin repo is cloned in test environment
+      ga_repo = Gitolite::GitoliteAdmin.new("#{::Rails.root.to_s}/data/test/repos/gitolite-admin", GITOLITE_SETTINGS)
+
+      # Bring the repo up to date
+      ga_repo.reload!
+
+      # Grab the gitolite admin repo config
+      conf = ga_repo.config
+
+      # Remove test repo from Gitolite conf
+      conf.rm_repo("test_repo")
+      
+      # Make sure repo was deleted, then remake it
+      repo = ga_repo.config.get_repo("test_repo")
+      if !repo.nil?
+        raise "Gitolite failed to delete the test_repo before test context!"
+      end
+
+      # Generate new test repo
+      repo = Gitolite::Config::Repo.new("test_repo")
+
+      # Add permissions for git user
+      repo.add_permission('RW+', '', 'git')
+
+      # Add the repo to the gitolite admin config
+      conf.add_repo(repo)
+
+      # Readd the 'git' public key to the gitolite admin repo after changes
+      admin_key = Gitolite::SSHKey.from_file(
+        GITOLITE_SETTINGS[:public_key])
+      ga_repo.add_key(admin_key)
+
+      # Stage and push the changes to the gitolite admin repo
+      ga_repo.save_and_apply
+
+      # Remove workdir (cloned version of the test_repo from Gitolite)
+      FileUtils.rm_rf("#{::Rails.root.to_s}/data/test/workdir")
+
+      # Repo is created by gitolite, proceed to clone it in
+      # the repository storage location
+      cloned_repo = Git.clone('git@localhost:' + "test_repo", "#{::Rails.root.to_s}/data/test/workdir")
```

Comment:
Redundant use of `Object#to_s` in interpolation.<br>Useless assignment to variable - `cloned_repo`.<br>Line is too long. [103/80]<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.